mindmeld/resource_loader.py

```diff
@@ -537,131 +629,121 @@ def _update_query_file_dates(self, query_tree):
+
+        def add(self, query):
+            for length, threshold in zip(self.lengths, self.thresholds):
+                if threshold > 0:
                     character_tokens = [
-                        q.normalized_text[i : i + length]
-                        for i in range(len(q.normalized_text))
-                        if len(q.normalized_text[i : i + length]) == length
+                        query.normalized_text[i : i + length]
+                        for i in range(len(query.normalized_text))
+                        if len(query.normalized_text[i : i + length]) == length
                     ]
-                    char_freq_dict.update(character_tokens)
-        return char_freq_dict
-
-    def _build_word_ngram_freq_dict(self, **kwargs):  # pylint: disable=no-self-use
-        """Compiles n-gram frequency dictionary of normalized query tokens
-
-        Args:
-            queries (list of Query): A list of all queries
-        """
-        enable_stemming = kwargs.get(ENABLE_STEMMING_ARGS)
-        word_freq_dict = Counter()
-        for length, threshold in zip(kwargs.get("lengths"), kwargs.get("thresholds")):
-            if threshold > 0:
-                ngram_tokens = []
-                for query in kwargs.get("queries"):
+                    self.char_freq_dict.update(character_tokens)
+
+        def get_resource(self):
+            return self.char_freq_dict
+
+    class WordNgramFreqBuilder:
+        '''
+        Compiles n-gram frequency dictionary of normalized query tokens
+        '''
+        def __init__(self, lengths, thresholds, enable_stemming=False):
+            self.lengths = lengths
+            self.thresholds = thresholds
+            self.enable_stemming = enable_stemming
+            self.word_freq_dict = Counter()
+
+        def add(self, query):
+            for length, threshold in zip(self.lengths, self.thresholds):
+                if threshold > 0:
+                    ngram_tokens = []
                     for i in range(len(query.normalized_tokens)):
                         ngram_query = " ".join(query.normalized_tokens[i : i + length])
                         ngram_tokens.append(ngram_query)
-                        if enable_stemming:
+                        if self.enable_stemming:
                             stemmed_ngram_query = " ".join(
                                 query.stemmed_tokens[i : i + length]
                             )
                             if stemmed_ngram_query != ngram_query:
                                 ngram_tokens.append(stemmed_ngram_query)
-                word_freq_dict.update(ngram_tokens)
-        return word_freq_dict
+                    self.word_freq_dict.update(ngram_tokens)
 
-    def _build_query_freq_dict(self, **kwargs):  # pylint: disable=no-self-use
-        """Compiles frequency dictionary of normalized and stemmed query strings
+        def get_resource(self):
+            return self.word_freq_dict
```

Comment:
wouldnt this also need a counter?

Reply:
self.work_freq_dict is a `Counter`.  Maybe I should change the name of the variable.  I just kept them the same as what they were.

Reply:
No worries, this is fine.